File: C++/binarySearchFromTo.cpp
#include <iostream>

int binarySearchFromTo(double elements[], double key, int from, int to){
    int low = from;
    int high = to;
    while (low <= high)
    {
        int mid = (low + high) / 2;
        double midVal = elements[mid];

        if (midVal < key)
        {
            low = mid + 1;
        } else{
            if (midVal > key)
            {
                high = mid -1;
            } else
            {
                return mid;
            }          
        }
    }
    return - (low +1);
}

File: Methods/Methods-Python/binarySearchFromTo.py
def binarySearchFromTo(elements, key, froom, to):

    low = froom
    high = to

    while low <= high:
        mid = (low + high)/2
        midVal = elements[mid]

        if midVal < key:
            low = mid + 1

        else:
            if midVal > key:
                high = mid - 1
            else:
                return mid
    return -(low + 1)

